sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jdbc/JdbcPropertiesBeanPostProcessor.java

```diff
@@ -134,20 +140,13 @@ private void enhanceUserAgent(DatabaseType databaseType, JdbcConnectionStringEnh
         }
     }
 
-    private Map<String, String> buildEnhancedProperties(DatabaseType databaseType, AzureJdbcPasswordlessProperties properties) {
+    private Map<String, String> buildEnhancedProperties(String passwordlessPropertiesPrefix,
+                                                        DatabaseType databaseType,
+                                                        AzureJdbcPasswordlessProperties properties) {
+        registerTokenCredentialBean(applicationContext, passwordlessPropertiesPrefix, properties);
```

Comment:
Same question here, if we can construct the token credential based on the proeprties, why do we need to register the bean here?